src/main/java/org/terasology/logic/health/HealthAuthoritySystem.java

```diff
@@ -20,13 +20,21 @@
     @In
     NUIManager nuiManager;
 
+    /**
+     * Sends out an immutable notification event when maxHealth of a character is changed.
+     */
     @ReceiveEvent(priority = EventPriority.PRIORITY_TRIVIAL)
     public void changeMaxHealth(ChangeMaxHealthEvent event, EntityRef player, HealthComponent health) {
         player.send(new OnMaxHealthChangedEvent(health.maxHealth, (int) event.getResultValue()));
         health.maxHealth = (int) event.getResultValue();
+        health.currentHealth = Math.min(health.currentHealth, health.maxHealth);
```

Comment:
1. please first change the values and send out the notification(s) afterwards
2. if current health is changed, we need to send out a notification for that too